django_socio_grpc/protobuf/proto_classes.py

```diff
@@ -228,6 +252,85 @@ def from_field(
             comments=comments,
         )
 
+    @classmethod
+    def get_field_comments(cls, field: serializers.Field):
+        if isinstance(help_text := getattr(field, "help_text", None), ProtoComment):
+            return help_text.comments
+        return None
+
+    @classmethod
+    def _from_choice_field(cls, field: serializers.ChoiceField):
+        if enum := cls.get_enum_from_annotation(field):
+            # Check if the annotated enum is a subclass of models.IntegerChoices or models.TextChoices
+            if not issubclass(enum, models.IntegerChoices) and not issubclass(
+                enum, models.TextChoices
+            ):
+                raise ProtoRegistrationError(
+                    f"Choice ({field.field_name}) field should be annotated with a class that is a subclass of models.IntegerChoices or models.TextChoices."
+                )
+
+            # Check for coherence between the choices and the annotated enum
+            if set(field.choices) != set(enum.values):
+                raise ProtoRegistrationError(
+                    f"Choice ({field.field_name}) field should be annotated with the same class as the choices, either on the serializer or the model.\n"
+                    "Example : `my_field : Annotated[models.CharField, MyTextChoices] = models.CharField(choices=MyTextChoices)`"
+                )
+
+            field_type = enum
+        elif cls._choice_field_contain_buildable_enum(field):
+            field_type = cls._build_enum_from_choice_field(field)
+        else:
+            field_type = get_proto_type(field)
+
+        return cls(
+            name=field.field_name,
+            field_type=field_type,
+            cardinality=cls._get_cardinality(field),
+            comments=cls.get_field_comments(field),
+        )
+
+    @classmethod
+    def _choice_field_contain_buildable_enum(cls, field: serializers.ChoiceField) -> bool:
```

Comment:
We can have two safe nets :
- setting to only generate enums from non annotated fields on new fields
- if a non annotated ChoiceField, has something else than a `string` type in the .proto (-> an enum), if it's no longer generable in the plugin, we raise an error

With this and the fact that we can choose to use or not the generation plugins, it should cover all possibilities ?

Reply:
And I don't understand why it has to be. 
As librairy maintener we need to provide the more easiest default behavior. 
Maybe we need to change some internal logic to make it not ambiguous

Reply:
it is ambiguous by nature, it could or could not generate an Enum depending on how it is defined.
But for me part of this ambiguity could be resolved by allowing to go from a string to a enum, but blocking going back to a string from an enum. This could be done by comparing to the previous protofile and raising an error.

For existing enums stored as string we can't raise an error which ask to modify the values so they are key-like because this would mean we have to fix all existing enums across all apps. too much of a breaking change. So still a bit of ambiguity there

Reply:
If there is ambiguity then free for the user to create its own, but too risky to have it implemented

Reply:
okay for me if we keep it simple as well, make the implementation simple as it remove the  need to get the previous protofile

Reply:
Why not fixing them automatically ? 
If it's just the internal mapping between django and grpc that create this ambiguous situation we can store in memory our own mapping and make it work everytime, isn't it ?

Reply:
good idea, but in the proto_serializer data_to_message and message_to_data we would need to have access to the plugin to know if we want automatic generation and the protofield to get access to the enum, else we need to build it dynamically again with each request

Reply:
yeah we'd like this to be finished + updated on all apps for tuesday, so a bit short on time, if we have an implementation for accessing protomessage in serializer, adding this feature of building enum for choices that are not already well formated would take little time, it's something like 10 lines of codes. So I think we might as well wait until we have the good architecture ?

Reply:
Okay for me